opennms-webapp-rest/src/main/java/org/opennms/web/rest/support/CriteriaBehaviors.java

```diff
@@ -0,0 +1,208 @@
+
+        EVENT_BEHAVIORS.put(Aliases.event.prop("eventAckTime"), new CriteriaBehavior<Date>(CriteriaBehaviors::parseDate));
+        EVENT_BEHAVIORS.put(Aliases.event.prop("eventCreateTime"), new CriteriaBehavior<Date>(CriteriaBehaviors::parseDate));
+        EVENT_BEHAVIORS.put(Aliases.event.prop("eventSeverity"), new CriteriaBehavior<Integer>(Integer::parseInt));
+        EVENT_BEHAVIORS.put(Aliases.event.prop("eventSuppressedCount"), new CriteriaBehavior<Integer>(Integer::parseInt));
+        EVENT_BEHAVIORS.put(Aliases.event.prop("eventTime"), new CriteriaBehavior<Date>(CriteriaBehaviors::parseDate));
+        EVENT_BEHAVIORS.put(Aliases.event.prop("eventTTicketState"), new CriteriaBehavior<Integer>(Integer::parseInt));
+        EVENT_BEHAVIORS.put(Aliases.event.prop("id"), new CriteriaBehavior<Integer>(Integer::parseInt));
+        EVENT_BEHAVIORS.put(Aliases.event.prop("ifIndex"), new CriteriaBehavior<Integer>(Integer::parseInt));
+        EVENT_BEHAVIORS.put(Aliases.event.prop("ipAddr"), new CriteriaBehavior<InetAddress>(InetAddressUtils::addr));
+
+        IP_INTERFACE_BEHAVIORS.put(Aliases.ipInterface.prop("id"), new CriteriaBehavior<Integer>(Integer::parseInt));
+        IP_INTERFACE_BEHAVIORS.put(Aliases.ipInterface.prop("lastCapsdPoll"), new CriteriaBehavior<Date>(CriteriaBehaviors::parseDate));
+        IP_INTERFACE_BEHAVIORS.put(Aliases.ipInterface.prop("ipAddress"), new CriteriaBehavior<InetAddress>(InetAddressUtils::addr));
+
+        MONITORING_LOCATION_BEHAVIORS.put(Aliases.location.prop("latitude"), new CriteriaBehavior<Float>(Float::parseFloat));
+        MONITORING_LOCATION_BEHAVIORS.put(Aliases.location.prop("longitude"), new CriteriaBehavior<Float>(Float::parseFloat));
+        MONITORING_LOCATION_BEHAVIORS.put(Aliases.location.prop("priority"), new CriteriaBehavior<Long>(Long::parseLong));
+        //MONITORING_LOCATION_BEHAVIORS.put(Aliases.location.prop("tags"), ???);
+
+        NODE_BEHAVIORS.put(Aliases.node.prop("id"), new CriteriaBehavior<Integer>(Integer::parseInt));
+        NODE_BEHAVIORS.put(Aliases.node.prop("createTime"), new CriteriaBehavior<Date>(CriteriaBehaviors::parseDate));
+
+        // Add aliases with join conditions when joining in the many-to-many node-to-category relationship
+        CriteriaBehavior<Integer> categoryId = new CriteriaBehavior<Integer>(Aliases.category.prop("id"), Integer::parseInt, (b,v,c,w) -> {
+            // Add a categories alias that only matches the specified value
+            // TODO: This should work but Hibernate is generating invalid SQL for this criteria
+            //b.alias(Aliases.node.prop("categories"), Aliases.category.toString(), JoinType.LEFT_JOIN, Restrictions.or(Restrictions.eq(Aliases.category.prop("id"), v), Restrictions.isNull(Aliases.category.prop("id"))));
+
+            switch (c) {
+            case EQUALS:
+                // TODO: Use parameterized statements to avoid SQL injection 
```

Comment:
Yes, this will require enhancing the Criteria API's Restrictions but should it should be easy to do.